packages/dapi/lib/grpcServer/handlers/blockheaders-stream/subscribeToBlockHeadersWithChainLocksHandlerFactory.js

```diff
@@ -126,19 +126,19 @@ function subscribeToBlockHeadersWithChainLocksHandlerFactory(
     let fromBlock;
 
     try {
-      // TODO: rework with getBlockStats
-      fromBlock = await coreAPI.getBlock(fromBlockHash);
+      fromBlock = await coreAPI.getBlockStats(fromBlockHash, ['height']);
```

Comment:
Since `getBlockStats` accepts height as well we can reduce the additional call for block hash above. So you can just pass hash or height here and ask not just for height but for block hash as well.

Reply:
@pshenmic and I also think you can use the height obtained from the block stats, and use it [here](https://github.com/dashevo/platform/blob/v0.22-dev/packages/dapi/lib/grpcServer/handlers/tx-filter-stream/subscribeToTransactionsWithProofsHandlerFactory.js#L218) and [here](https://github.com/dashevo/platform/blob/v0.22-dev/packages/dapi/lib/grpcServer/handlers/blockheaders-stream/subscribeToBlockHeadersWithChainLocksHandlerFactory.js#L168) instead of the fromBlockHash